### PR TITLE
[#2dmz3hj] Header

### PR DESCRIPTION
**Pull request by vstacked**
## Description
Adding header widget using assets svg for image & icon.
- add ui for portrait & landscape (base on scratch image on clickUp)

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/page/home/widget/header.dart [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  const Header({Key? key, this.rating = 4.9, this.sales = 542, this.imageName = 'drone_1.png', this.onPressed})
+      : super(key: key);
+  final double rating;
+  final int sales;
+  final String imageName;
+  final VoidCallback? onPressed;
+
+  @override
+  Widget build(BuildContext context) {
+    final width = MediaQuery.of(context).size.width;
```
Comment (nfach98) [inline lib/page/home/widget/header.dart]: Opsional, untuk width bisa dijadikan parameter widget

**Review by nfach98 — APPROVED** [review]

**Review by isa9x — APPROVED** [review]

**Review by StanislausPN — APPROVED** [review]

**Review by StanislausPN — APPROVED** [review]

File: lib/page/home/widget/header.dart [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  }) : super(key: key);
+  final double rating;
+  final int sales;
+  final String imageName;
+  final VoidCallback? onPressed;
+  final double? width;
+
+  @override
+  Widget build(BuildContext context) {
+    final width = this.width ?? MediaQuery.of(context).size.width;
```
Comment (mnafian) [inline lib/page/home/widget/header.dart]: karena MediaQuery di pake di banyak tempat, consider dibuat ext seperti yg aku contohkan keamaren biar tidak terlalu banyak deklarasi ulang

File: lib/page/home/widget/header.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          ),
+          SizedBox(
+            // TODO need padding default, dummy 32
+            width: (width - 32) / 2,
+            child: Padding(
+              padding: const EdgeInsets.all(20.0),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                mainAxisAlignment: MainAxisAlignment.end,
+                children: <Widget>[
```
Comment (mnafian) [inline lib/page/home/widget/header.dart]: opsional, di sdk existing yg dipake harusnya secara implisit tanpa declare ```<widget>``` udah bisa, ini dihapus aja sama yg dibawah2

**Review by mnafian — CHANGES_REQUESTED** [review]

File: lib/page/home/widget/header.dart (hunk 3 above)
Comment (vstacked) [reply lib/page/home/widget/header.dart]: berarti ngga pake `crossAxisAlignment` sama `mainAxisAlignment` mas? itu saya pake buat positioning text & button widget biar kebawah

Comment (vstacked) [reply lib/page/home/widget/header.dart]: ohh baik mas

Comment (mnafian) [reply lib/page/home/widget/header.dart]: updated comment ya, somehow kalo aku nulis `<widget>` gak bisa di render sama comment github kalo gak format as markdown, maksudnya hapus widget declarationnya aja

**Review by mnafian — APPROVED** [review]

File: lib/page/home/widget/header.dart (hunk 3 above)
Comment (vstacked) [reply lib/page/home/widget/header.dart]: oalahh, iya mas

**Pull request merged**